### PR TITLE
TPriorityList bug fix with unit test, TApplication::getModulesByType correction, TEventContent & TPageLoadTime (addition with unit test)

### PR DESCRIPTION
**Pull request by belisoful**
TPriorityList bug - cache did not update when exists but empty, fixed with unit test, caching performance speed-up

TApplication::getModulesByType - when setting modules to null, null modules may not be a sign of lazy load modules

TEventContent is an experimental control for allowing composer PRADO package modules to fill in controls, like navigation.  A global event is raised and the handlers fill in controls (like list items).

TPageLoadTime measures the time from starting the request to the tag render time.

### CODE REVIEW

File: framework/Util/TDbParameterModule.php [hunk 1]
```diff
@@ -233,14 +233,19 @@ protected function createDbTable()
 		$db = $this->getDbConnection();
 		$driver = $db->getDriverName();
 		$autoidAttributes = '';
+		$autotype = 'INTEGER';
 		$postIndices = '; CREATE UNIQUE INDEX tkey ON ' . $this->_tableName . '(' . $this->_keyField . ');' .
 		($this->_autoLoadField ? ' CREATE INDEX tauto ON ' . $this->_tableName . '(' . $this->_autoLoadField . ');' : '');
 		if ($driver === 'mysql') {
```
Comment (ctrlaltca) [inline framework/Util/TDbParameterModule.php:239]: use switch() instead?
The default branch could either default to mysql or throw an exception.

File: framework/Web/UI/WebControls/TPageLoadTime.php [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	 */
+	public function renderContents($writer)
+	{
+		$writer->write(round(microtime(true) - $_SERVER["REQUEST_TIME_FLOAT"], 5) . Prado::localize($this->getSecondAppendix()));
+	}
+	
+	/**
+	 * @return string the string that is appended to the time.  default 's' for seconds.
+	 */
+	public function getSecondAppendix()
```
Comment (ctrlaltca) [inline framework/Web/UI/WebControls/TPageLoadTime.php:42]: Suffix?

**Review by ctrlaltca — APPROVED** [review]
Looks good! two minor comments, not blocking.

File: framework/Util/TDbParameterModule.php (hunk 1 above)
Comment (belisoful) [reply framework/Util/TDbParameterModule.php:239]: good point

File: framework/Web/UI/WebControls/TPageLoadTime.php (hunk 2 above)
Comment (belisoful) [reply framework/Web/UI/WebControls/TPageLoadTime.php]: changed

File: framework/Util/TDbParameterModule.php (hunk 1 above)
Comment (belisoful) [reply framework/Util/TDbParameterModule.php]: corrected

**Pull request merged**